presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -209,7 +210,10 @@ public ActualProperties visitWindow(WindowNode node, List<ActualProperties> inpu
             ActualProperties properties = Iterables.getOnlyElement(inputProperties);
 
             // If the input is completely pre-partitioned and sorted, then the original input properties will be respected
-            if (ImmutableSet.copyOf(node.getPartitionBy()).equals(node.getPrePartitionedInputs()) && node.getPreSortedOrderPrefix() == node.getOrderBy().size()) {
+            Optional<OrderingScheme> orderingScheme = node.getOrderingScheme();
+            if (ImmutableSet.copyOf(node.getPartitionBy()).equals(node.getPrePartitionedInputs())
+                    && orderingScheme.isPresent()
+                    && node.getPreSortedOrderPrefix() == orderingScheme.get().getOrderBy().size()) {
```

Comment:
the code you are changing looks looks like a hack

Reply:
TBH it probably requires to understand window operator more deeply. For now I will preserver previous semantics:
```
node.getPreSortedOrderPrefix() == node.getOrderBy().size())
```
which translates to:
```
node.getPreSortedOrderPrefix() == 0 || node.getPreSortedOrderPrefix() == orderingScheme.get().getOrderBy().size()
```